cEP-0020.md

```diff
@@ -0,0 +1,389 @@
+- Badge-I:
+  - Name: The bug finder
+  - Details: The one who find bugs in the existing codebase
+  - Activities:
+    - Create at least one issue which labeled as bug
+      and approved by a maintainer i.e.: had label `process/verified`
+- Badge-II:
+  - Name: The bear hunter
+  - Details: The one who create issues about new bears
+  - Activities:
+    - Create at least one bear proposal issue that also had label
+      `process/varified`
+- Badge-III:
+  - Name: The bear writer
+  - Details: The one who write new bears
+  - Activities:
+    - Merged at least 2 bear proposal PRs
+- Badge-IV:
+  - Name: The helper
+  - Details: The one who answers every newcomer questions on Gitter
+  - Activities:
+    - At least 20 good messages to newcomer
+- Badge-V:
+  - Name: The coala Developer
+  - Details: The ones who are in the coala developers team
+  - Activities:
+    - Introduce `.coafile` in other projects
+    - Merge a difficulty/newcomer Pull Request
+    - Review at least a difficulty/newcomer Pull Request
+    - Merge a difficulty/low Pull Request
+    - Review at least a difficulty/low or higher Pull Request
+
+So, for becoming a developer at coala, a newcomer must complete all the
+activities mentioned in the `coala Developer` badge. But in case a newcomer
+is not able to become a developer at coala, he/she could take away the other
+badges.
+
+#### Note
+
+Names and metrics of tracked activities will also be shown on the user
+dashboard along with the number of times they have been done. That will
+help maintainers to check if an automatically awarded badge is correct or
+not. So, in case of doubts on any newcomer, there will be a button to
+revoke those badges.
+
+The `coala Developer` badge doesn't mean that a newcomer is in the coala
+developer team. If a newcomer wants to be the part of the developer team
+after earning the developer badges, he/she will have to ask a maintainer
+on Gitter to verify all the activities they have done, on the dashboard.
+
+## Implementation
+
+### Getting the data
+
+#### GitHub/GitLab
+
+For getting the data related to newcomers activity on GitHub/GitLab to the
+community, we will mainly use [IGitt](https://gitlab.com/gitmate/open-source/IGitt),
+But for getting the data related to review comments on PRs we will use the
+data provided by the [meta-review project](https://github.com/coala/cEPs/blob/e9f7587856e1c236d37e96f54a0cb4d72069776d/cEP-0019.md).
+
+#### Gitter
+
+For getting the stats of newcomer's messages, we will use
+[Gitter API](https://developer.gitter.im/docs/messages-resource) and import
+all the messages to newcomers or by newcomers in our webservices. And then
+we will import those message to our community repos into two Django models
+i.e.: `GoodMessage` and `BadMessage`
+These messages can be analysed and categorized with the use of
+[basic sentiment analysis](http://fjavieralba.com/basic-sentiment-analysis-with-python.html)
+
+### A System for Tracking Activities, Points and Levels
```

Comment:
How does this system work in general? Does it fetch data periodically (cron job maybe), process the data and display them somewhere (dashboard?)?

Reply:
Yeah, There will be Django management commands which will be triggered on every build on netlify.
>display them somewhere (dashboard?)?
Yeah, there will also be a user dasboard which will show stats and metrics of each user.

Reply:
Added!